app/src/main/java/com/example/drawingApp/SelectedColorView.kt

```diff
@@ -0,0 +1,86 @@
+package com.example.drawingApp
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+
+
+@SuppressLint("ClickableViewAccessibility")
+/**
+ * This class is ued to create a custom view
+ * the view shows a solid circle
+ * when clicked it will cycle forward in its color order
+ * the color order is Black, Red, Green, Blue
+ *
+ */
+class SelectedColorView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    private var states = arrayOf(Color.BLACK,Color.RED,Color.GREEN,Color.BLUE)
+    private var state = states[0]
+    private val paint = Paint()
+
+    /**
+     * pass the selected color and assign it to given variables
+     * use getState function from the SelectedColorView object to get the color
+     * will be called everytime the color is changed
+     */
+    var passColor: (() -> Unit)? = null
```

Comment:
I still don't understand this. It's not giving anything into the lambda. This lambda should be self contained. Why make someone reference the view itself? And a callback should be named with `on` at the start usually for what triggers it

Reply:
realized that the get state function is irrelevant, your right the view shouldn't need to be referenced, just having it pass an int, and when invoked passing it the state variable which holds the int for the color